src/menhir_parser.mly

```diff
@@ -59,31 +60,30 @@ let translate_old_rule : old_p_rule -> p_rule = fun r ->
             begin
               try
                 let n = Hashtbl.find arity x in
-                if nb_args < n then Hashtbl.replace arity x nb_args
+                if nb_args > n then Hashtbl.replace arity x nb_args
               with Not_found -> Hashtbl.add arity x nb_args
             end
       | P_Wild          -> ()
-      | P_Type          -> ()
+      | P_Type          -> fatal h.pos "Type in legacy pattern."
       | P_Prod(_,_)     -> fatal h.pos "Product in legacy pattern."
-      | P_Meta(_,_)     -> fatal h.pos "Metaviable in legacy pattern."
       | P_Abst(xs,t)    ->
           begin
             match xs with
             | [(_       ,Some(a),_)] ->
                 fatal a.pos "Annotation in legacy pattern."
             | [([Some x],None   ,_)] ->
                 compute_arities (x.elt::env) t
-            | _                      ->
-                fatal h.pos "Invalid legacy pattern lambda."
+            | _                      -> assert false
           end
-      | P_Patt(_,_)     -> fatal h.pos "Pattern in legacy rule."
       | P_Impl(_,_)     -> fatal h.pos "Implication in legacy pattern."
-      | P_LLet(_,_,_,_) -> fatal h.pos "Let expression in legacy rule."
```

Comment:
There is a let in legacy Dedukti, so perhaps we should keep the error message for it

Reply:
fixed in c73b6ac